lib/ansible/module_utils/oneview.py

```diff
@@ -0,0 +1,440 @@
+# This code is part of Ansible, but is an independent component.
+# This particular file snippet, and this file snippet only, is BSD licensed.
+# Modules you write using this snippet, which is embedded dynamically by Ansible
+# still belong to the author of the module, and may assign their own license
+# to the complete work.
+#
+# Copyright (2016-2017) Hewlett Packard Enterprise Development LP
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#
+#    * Redistributions of source code must retain the above copyright
+#      notice, this list of conditions and the following disclaimer.
+#    * Redistributions in binary form must reproduce the above copyright notice,
+#      this list of conditions and the following disclaimer in the documentation
+#      and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import (absolute_import, division, print_function)
+
+import abc
+import collections
+import json
+import logging
+import os
+import traceback
+from copy import deepcopy
+
+from ansible.module_utils import six
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+try:
+    from hpOneView.oneview_client import OneViewClient
+    from hpOneView.exceptions import (HPOneViewException,
+                                      HPOneViewTaskError,
+                                      HPOneViewValueError,
+                                      HPOneViewResourceNotFound)
+
+    HAS_HPE_ONEVIEW = True
+except ImportError:
+    HAS_HPE_ONEVIEW = False
+
+
+def transform_list_to_dict(list_):
+    """
+    Transforms a list into a dictionary, putting values as keys.
+
+    :arg list list_: List of values
+    :return: dict: dictionary built
+"""
+
+    ret = {}
+
+    if not list_:
+        return ret
+
+    for value in list_:
+        if isinstance(value, collections.Mapping):
+            ret.update(value)
+        else:
+            ret[to_native(value, errors='surrogate_or_strict')] = True
+
+    return ret
+
+
+class OneViewModuleBase(object):
+    __metaclass__ = abc.ABCMeta
```

Comment:
Oh sorry, I should have mentioned, how to specify metaclasses changed between python2 and python3.  So there's a helper in six so that you can do this acroos both versions.
```
from ansible.module_utils.six import with_metaclass
[...]
class OneViewModuleBase(with_metaclass(ABCMeta, object)):
```